dissect/target/plugins/filesystem/unix/capability.py

```diff
@@ -82,88 +82,104 @@ class CapabilityPlugin(Plugin):
-                        data_len = (1 + 2 * VFS_CAP_U32_1) * 4
-                    elif cap_revision == VFS_CAP_REVISION_2:
-                        num_caps = VFS_CAP_U32_2
-                        data_len = (1 + 2 * VFS_CAP_U32_2) * 4
-                    elif cap_revision == VFS_CAP_REVISION_3:
-                        num_caps = VFS_CAP_U32_3
-                        data_len = (2 + 2 * VFS_CAP_U32_2) * 4
-                    else:
-                        self.target.log.error("Unexpected capability revision: %s", entry)
-                        continue
-
-                    if data_len != len(attr.value):
-                        self.target.log.error("Unexpected capability length: %s", entry)
-                        continue
-
-                    for _ in range(num_caps):
-                        permitted_val, inheritable_val = struct.unpack("<2I", buf.read(8))
-                        permitted_caps.append(permitted_val)
-                        inheritable_caps.append(inheritable_val)
-
-                    if cap_revision == VFS_CAP_REVISION_3:
-                        rootid = struct.unpack("<I", buf.read(4))[0]
-
-                    permitted = []
-                    inheritable = []
-
-                    for capability in Capabilities:
-                        for caps, results in [(permitted_caps, permitted), (inheritable_caps, inheritable)]:
-                            # CAP_TO_INDEX
-                            cap_index = capability.value >> 5
-                            if cap_index >= len(caps):
-                                # We loop over all capabilities, but might only have a version 1 caps list
-                                continue
-
-                            if caps[cap_index] & (1 << (capability.value & 31)) != 0:
-                                results.append(capability.name)
-
-                    yield CapabilityRecord(
-                        record=record,
-                        permitted=permitted,
-                        inheritable=inheritable,
-                        effective=magic_etc & VFS_CAP_FLAGS_EFFECTIVE != 0,
-                        rootid=rootid,
-                        _target=self.target,
-                    )
+                    permitted, inheritable, effective, root_id = parse_attr(attr.value)
+                except ValueError as e:
+                    self.target.log.warning("Could not parse attributes for entry %s: %s", entry, str(e.value))
+                    self.target.log.debug("", exc_info=e)
+
+                yield CapabilityRecord(
+                    ts_mtime=entry.lstat().st_mtime,
+                    path=entry.path,
+                    permitted=permitted,
+                    inheritable=inheritable,
+                    effective=effective,
+                    root_id=root_id,
+                    _target=self.target,
+                )
+
+
+def parse_attr(attr: bytes) -> tuple[int, list[str], list[str], bool]:
```

Comment:
```suggestion
def parse_attr(attr: bytes) -> tuple[list[str], list[str], bool, int]:
```
My bad, I changed the order later :smile:.